src/Symfony/Component/Validator/Constraints/ExpressionLanguageSyntaxValidator.php

```diff
@@ -42,6 +42,10 @@ public function validate($expression, Constraint $constraint): void
             throw new UnexpectedTypeException($expression, 'string');
         }
 
+        if (null === $this->expressionLanguage) {
+            $this->expressionLanguage = new ExpressionLanguage();
```

Comment:
This will break the ability to use allowed functions in expression. This can cause confusion among developers. See ExpressionLanguage constructor: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php#L37-L44

Reply:
I considered this option, and I'm not sure that it is correct.

Reply:
hmm. This is a fallback when user don't provide ExpressionLanguage instance in `__constuctor`.
People with specific `ExpressoinLanguage` still have the abaility to inject it in the Validator.

Reply:
restored the `ExpressionLanguageSyntax::service` to allow people using their own Validator

Reply:
Thanks, it looks like it will work!